exonum/src/runtime/mod.rs

```diff
@@ -649,10 +672,19 @@ impl<'a> ExecutionContext<'a> {
 
         // Add service instance to the dispatcher schema.
         DispatcherSchema::new(&*self.fork)
-            .add_pending_service(spec)
+            .initiate_adding_service(spec)
             .map_err(From::from)
     }
 
+    /// Initiates stopping of an existing service instance in the blockchain. The stopping
+    /// service is active (i.e., does process transactions or `after_transactions` hook)
+    /// until the block built on top of the provided `fork` is committed.
+    ///
+    /// This method should be called for the exact context passed to the runtime.
+    pub(crate) fn initiate_stopping_service(&mut self, instance_id: InstanceId) {
```

Comment:
Not sure I understand why this is placed in `ExecutionContext` or takes `self` by mutable ref. (Cf. `CallContext::start_artifact_registration`, which calls to a static method in `Dispatcher`.) For symmetry with `initiate_adding_service`? Since the method is private, I don't think future compatibility should apply.